File: src/Godot.cpp
//
// Created by cedric on 13.08.20.
//

#include "Godot.h"

godot_gdnative_ext_nativescript_api_struct *Godot::nativescript = nullptr;
const godot_gdnative_core_api_struct *Godot::gdnative = nullptr;

void Godot::init(godot_gdnative_init_options *options) {
    gdnative = options->api_struct;
    auto extensionCount = gdnative->num_extensions;
    auto extensions = gdnative->extensions;

    for (int i = 0; i < extensionCount; ++i) {
        auto extension = extensions[i];
        auto type = extension->type;
        if (type == 1) {
            nativescript = (godot_gdnative_ext_nativescript_api_struct*) (extension);
        }
    }
}

File: src/godot.rs
use gdnative::*;
use std::ptr;
use std::ffi::c_void;
use std::ptr::null;

struct GodotState {
    gdnative: *const godot_gdnative_core_api_struct
}

static mut GODOT_STATE: GodotState = GodotState {
    gdnative: null()
};

pub struct Godot;

pub fn init(options: *const godot_gdnative_init_options) {
    unsafe {
        GODOT_STATE.gdnative = (*options).api_struct;
    }
    println!("GDScript Initialized!")
}

pub fn nativescript_init(handle: *const c_void) {

}

pub fn nativescript_destroy(handle: *const c_void) {

}

pub fn terminate(options: *const godot_gdnative_terminate_options) {

}